Core/clim-core/frames.lisp

```diff
@@ -240,7 +241,9 @@ documentation produced by presentations.")
     (if (and (get-menu (frame-command-table frame))
 	     (get-menu new-command-table))
 	(prog1 (call-next-method)
-	  (generate-panes *default-frame-manager* *application-frame*))
+	  (when (slot-value frame 'menu-bar-pane)
+	    (update-menu-bar (slot-value frame 'menu-bar-pane)
+			     new-command-table)))
```

Comment:
if there is no possibility, that slot-value chagnes between condition check and call update-menu-bar, then please use

```
(alexandria:when-let ((menu-bar-pane (slot-value frame -menu-bar-pane)))
  (update-menu-bar menu-bar-pane …))
```